floatcounter.go

```diff
@@ -0,0 +1,82 @@
+package metrics
+
+import (
+	"fmt"
+	"io"
+	"sync"
+)
+
+// NewFloatCounter registers and returns new counter of float64 type with the given name.
+//
+// name must be valid Prometheus-compatible metric with possible labels.
+// For instance,
+//
+//     * foo
+//     * foo{bar="baz"}
+//     * foo{bar="baz",aaa="b"}
+//
+// The returned counter is safe to use from concurrent goroutines.
+func NewFloatCounter(name string) *FloatCounter {
+	return defaultSet.NewFloatCounter(name)
+}
+
+// FloatCounter is a float64 counter guarded by RWmutex.
+//
+// It may be used as a gauge if Add and Sub are called.
+type FloatCounter struct {
+	mu sync.RWMutex
+	n  float64
+}
+
+// Add adds n to fc.
+func (fc *FloatCounter) Add(n float64) {
+	fc.mu.Lock()
+	fc.n = fc.n + n
```

Comment:
This is usually written as `fc.n += n`

Reply:
fixed in a871ded987f5e4c3af37ed04cbae4d3551359b93